File: leetcode/java/0463_island_perimeter.java
class Solution {
    public int islandPerimeter(int[][] grid) {
        // loop through grid using i:
        // loop through grid[i] using j:
        // if grid[i][j] == 1:
        //  if top side has water or is at the edge: perimeter++
        //  if right side has water or is at th eedge: perimeter++ ....
        int perimeter = 0;
        
        for (int i = 0; i < grid.length; i++) {
            for (int j = 0; j < grid[i].length; j++) {
                if (grid[i][j] == 1) {
                    if (i == 0 || (grid[i - 1][j] == 0)) perimeter++;
                    if (j == grid[i].length - 1 || (grid[i][j + 1] == 0)) perimeter++;
                    if (i == grid.length - 1 || (grid[i + 1][j] == 0)) perimeter++;
                    if (j == 0 || (grid[i][j - 1] == 0)) perimeter++;
                }
            }
        }

        return perimeter;
    }
}

File: leetcode/python/0463_island_perimeter.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        res = 0
        for i in range(0, len(grid)):
            for j in range(0, len(grid[i])):
                if grid[i][j] == 1:
                    if i == 0 or grid[i - 1][j] == 0: res += 1
                    if j == 0 or grid[i][j - 1] == 0: res += 1
                    if i == len(grid) - 1 or grid[i + 1][j] == 0: res += 1
                    if j == len(grid[i]) - 1 or grid[i][j + 1] == 0: res += 1

        return res